report_presentation/Presenter.py

```diff
@@ -7,4 +16,274 @@ def __init__(self, report_root, initial_warnings, final_warnings, changes):
+        html = '''
+                        <p class="serif" ><b>     8.  Detailed Warning Information (Before CTA)</b></p>
+                        <div class = "before-warning-table">
+                        <table border = "0">
+                                <tr>
+                                        <th>No.</th>
+                                        <th>Warning Code</th>
+                                        <th>File Path</th>
+                                        <th>Project Name</th>
+                                        <th>Line Number</th>
+                                        <th>Warning message</th>
+                                </tr>'''
+        html += self.get_warning_info(all_warnings)
+        html += '''
+                        </table>
+                        </div>
+                        '''
+        return html
+
+    def get_html9(self, final_warnings):
+        html = '''
+                        <p class="serif" ><b>     9.  Detailed Warning Information (After CTA)</b></p>
+                        <div class = "after-warning-table">
+                        <table border = "0">
+                                <tr>
+                                        <th>No.</th>
+                                        <th>Warning Code</th>
+                                        <th>File Path</th>
+                                        <th>Project Name</th>
+                                        <th>Line Number</th>
+                                        <th>Warning message</th>
+                                </tr>'''
+        html += self.get_warning_info(final_warnings)
+        html += '''
+                        </table>
+                        </div>
+                        '''
+        return html
+
+    def get_html10(self, changes):
+        html = '''
+                        <p class="serif" ><b>     10.  Detailed Recommendation Information (After CTA)</b></p>
+                        <div class = "recommendation-table">
+                        <table border = "0">
+                                <tr>
+                                        <th>No.</th>
+                                        <th>Recommendation Code</th>
+                                        <th>File Path</th>
+                                        <th>Project Name</th>
+                                        <th>Line Number</th>
+                                        <th>Recommendation message</th>
+                                </tr>'''
+        html += self.get_recommendation_info(changes)
+        html += '''
+                        </table>
+                        </div>
+                        '''
+        return html
+
+    def get_html11_13(self, warning_fixed, changes, diff_path):
+        html =''
+        html += '''
+                        <p class="serif" ><b>     11. Number of warnings have been fixed: %s</b></p>
+                                ''' % warning_fixed
```

Comment:
here and in other places with strings, is there a reason why there's so much indentation? Btw depending on your pycharm configuration, usually you can auto-fix indentation of a file with cmd+ctr+L